### ISSUE TITLE
Several units with unmet start conditions rate limited on startup

### ISSUE THREAD

**Issue by falconindy** [systemd/systemd#2684]
Downstream bug report: https://bugs.archlinux.org/task/48267

Something with job merging not working correctly in 229? I don't really know... I've attached a [debug log](https://github.com/systemd/systemd/files/139807/log.txt.gz) which shows services, (e.g. ldconfig) being repeatedly requested to start and eventually getting rate limited.

**Comment by poettering** [comment]
hmm, any idea how I can reproduce this here locally?

**Comment by falconindy** [comment]
Looks like I can trigger this by setting `DefaultStartLimitBurst=1` in /etc/systemd/system.conf.

**Comment by falconindy** [comment]
Maybe this is a result of 6bf0f408e483315219?

**Comment by falconindy** [comment]
Seems like the culprit. Reverting it on top of 229 (along with some unrelated bugfixes) makes the rate limit errors go away. I suppose that makes sense, since these units then no longer get an implicit start limit applied.

Though, I guess the real problem is that the unit gets requeued instead of staying dead with the condition unmet.

**Comment by DeX77** [comment]
Like already mentioned in https://bugs.archlinux.org/task/48267 I can confirm that removing laptop-mode-tools fixes the issue.

So you can test it with this.

**Comment by falconindy** [comment]
Actually, it wasn't explicitly mentioned there -- I just asked the question because I was starting to suspect it was related. Now I'm convinced. The problem is [this script](https://github.com/rickysarraf/laptop-mode-tools/blob/lmt-upstream/etc/rules/lmt-udev), which is triggered by a rather broad udev rule which eventually calls `systemctl --no-block reload-or-restart laptop-mode`, perhaps multiple times during startup.

Awful behavior... I don't know if this is even something systemd should support.

**Comment by DeX77** [comment]
But how does triggering a laptop-mode restart lead to those failures? All those services are using a condition on /etc or similar.

**Comment by wangjiezhe** [comment]
I upgrade laptop-mode to the newest git version and there is no failure now.

**Comment by DeX77** [comment]
Weird. Did the same and still see:

> systemd[1]: Failed to mount Configuration File System.
> systemd[1]: Failed to mount Virtual Machine and Container Storage.
> systemd[1]: Failed to start First Boot Wizard.

But only once this time.

**Comment by starquake** [comment]
I upgraded to the newest git version of laptop-mode and also fixed it for me

**Comment by drrossum** [comment]
I am getting this error without laptop-mode-tools installed.  A systemd.log_level=debug log is attached.

**Comment by drrossum** [comment]
[debug.txt](https://github.com/systemd/systemd/files/158916/debug.txt)

**Comment by benjarobin** [comment]
To reproduce this issue just create these 2 files : 

> /usr/lib/udev/rules.d/99-testlmt.rules

```
ACTION=="add|remove", RUN+="/usr/bin/systemctl --no-block reload-or-restart testlmt"
```

> /etc/systemd/system/testlmt.service

```
[Unit]
Description=TESTLMT

[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=/usr/bin/sleep 0.1
ExecReload=/usr/bin/sleep 0.1
StandardOutput=journal
StandardError=journal

[Install]
WantedBy=multi-user.target
```

**Comment by poettering** [comment]
Fix waiting in #3148.
